azure/services/natgateways/natgateways.go

```diff
@@ -99,7 +100,15 @@ func (s *Service) Reconcile(ctx context.Context) error {
 			return errors.Wrapf(err, "failed to create nat gateway %s in resource group %s", natGatewaySpec.Name, s.Scope.ResourceGroup())
 		}
 		s.Scope.V(2).Info("successfully created nat gateway", "nat gateway", natGatewaySpec.Name)
-		s.Scope.SetNodeNatGatewayID(azure.NatGatewayID(s.Scope.SubscriptionID(), s.Scope.ResourceGroup(), natGatewaySpec.Name))
+		natGateway := infrav1.NatGateway{
+			ID:   azure.NatGatewayID(s.Scope.SubscriptionID(), s.Scope.ResourceGroup(), natGatewaySpec.Name),
+			Name: natGatewaySpec.Name,
+			NatGatewayIP: infrav1.PublicIPSpec{
+				Name: *(*natGatewayToCreate.NatGatewayPropertiesFormat.PublicIPAddresses)[0].ID,
```

Comment:
We'll either need to create a new field `ID` for  public ip spec, or not set it. There was an issue reported for this: https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1525

Reply:
I changed it so that it uses the name instead of the ID.